src/main/java/org/elasticsearch/cluster/metadata/MetaData.java

```diff
@@ -675,6 +675,14 @@ public String resolveIndexRouting(@Nullable String routing, String aliasOrIndex)
 
             aliasesOrIndices = convertFromWildcards(aliasesOrIndices, indicesOptions);
         }
+
+        if (isAllIndices(aliasesOrIndices)) {
+            if (!indicesOptions.allowNoIndices()) {
+                throw new IndexMissingException(new Index("_all"));
+            }
+            else return Strings.EMPTY_ARRAY;
+        }
+
```

Comment:
can you explain what you are trying to achieve with this code block? Does it actually change the result of calling `concreteIndices`? Seems like it doesn't, il wildcards don't need to be expanded _all is treated as a concrete index, but it's not there, then we look at allowNoIndices anyway to decide whether we want to throw exception or just return empty. What am I missing?